### DISCUSSION TITLE
Different scripts for different pages?

### DISCUSSION THREAD

**Discussion by HEIGE-PCloud** [PaperStrike/Pjax#46] [Q&A] [answered]
A similar question https://github.com/MoOx/pjax/issues/230

I want to have different `<script src>` tags in different pages, how should I handle this?

For example, I have 

```html
<script src="pjax.js"></script>
<script src="theme.js"></script>
```

on page 1, and `Pjax()` is initialized in `theme.js`.

And I have

```html
<script src="lightgallery.js"></script>
<script src="pjax.js"></script>
<script src="theme.js"></script>
```

on page 2, which I want to call `if (!window.lightgallery) lightgallery.init()` in `theme.js` when the page is loaded. How should I organize the DOM tree and reload the function?

**Comment by PaperStrike** [comment, marked as answer]
The easiest solution would be wrapping the parts that differ among pages into an element that you will add to the `selectors` option. Make sure to have the wrapper existing on each page. In your example, that could be

```html
<!-- page 1 -->
<div class="differ-among-pages"></div>
<script src="pjax.js"></script>
<script src="theme.js"></script>

<!-- page 2 -->
<div class="differ-among-pages">
  <script src="lightgallery.js"></script>
</div>
<script src="pjax.js"></script>
<script src="theme.js"></script>
```

don't forget to select that element like:

```js
const pjax = new Pjax({
  selectors: [
    '.differ-among-pages',
  ],
});
```

https://github.com/PaperStrike/Pjax/discussions/31#discussioncomment-866715 shows two other possible solutions.
